whipper/src/main/java/org/whipper/resultmode/CompareResultMode.java

```diff
@@ -65,16 +66,27 @@ public ResultHolder handleResult(Query q){
         ResultHolder out = new ResultHolder();
         OverrideFileSelector selector = new OverrideFileSelector(q.getScenario().getExpectedResultsDir());
         File result = selector.getExpectedResultFile(getExpectedResultFileName(q));
-        String expectedResultDirectoryName = result.getParentFile().getParentFile().getName();
-        try{
-            holder.buildResult(result, q);
-            boolean eq = holder.equals(q.getActualResult(), !q.getSql().toUpperCase().contains(" ORDER BY "), allowedDivergence);
-            if(!eq){
-                writeErrorFile(q, expectedResultDirectoryName);
-                out.setErrors(holder.getErrors());
+        if (result != null && result.exists()) {
+            String expectedResultDirectoryName = result.getParentFile().getParentFile().getName();
+            try{
+                holder.buildResult(result, q);
+                boolean eq = holder.equals(q.getActualResult(), !q.getSql().toUpperCase().contains(" ORDER BY "), allowedDivergence);
+                if(!eq){
+                    writeErrorFile(q, expectedResultDirectoryName);
+                    out.setErrors(holder.getErrors());
+                }
+            } catch (IOException ex){
+                out.setException(ex);
+            }
+
+        } else {
+            holder.getErrors().add("Unable to find expected result file for query");
```

Comment:
This is useless, since you are adding error to previous comparison (holder.clear() is not called - this is private method called in method builResult()).

Reply:
Oh, I didn't notice how the holder is shared and mutable. I'll come up with a better solution after the weekend.